website/docs/language/tests/index.mdx

```diff
@@ -0,0 +1,685 @@
+    # assertions aren't important for this example.
+  }
+
+  # ... more assertions ...
+}
+
+run "update_setup" {
+
+  # We've now re-referenced the setup module, so the state file that was created
+  # for the first "setup" run block will be reused. It will contain any
+  # resources that were created as part of the other run block before this run
+  # block executes and will be updated with any changes made by this run block
+  # after.
+
+  module {
+    source = "./testing/setup"
+  }
+
+  variables {
+    # In practice, we'd likely make some changes to the module compared to the
+    # first run block here. Otherwise there would be no point recalling the
+    # module. However, the concrete changes aren't important to this example.
+  }
+}
+
+run "update" {
+
+  # As with the "init" run block we are executing against the main configuration
+  # again. This means we'd load the main state file that was initially populated
+  # by the "init" run block, and any changes made by this "run" block will be
+  # carried forward to any future run blocks that execute against the main
+  # configuration.
+
+  # ... updated variables ...
+
+  # ... assertions ...
+}
+
+run "loader" {
+
+  # This run block is now referencing our second alternate module so will create
+  # our third and final state file. The other two state files are managing
+  # resources from the main configuration and resources from the setup module.
+  # We are getting a new state file for this run block as the loader module has
+  # not previously been referenced by any run blocks.
+
+  module {
+    source = "./testing/loader"
+  }
+}
+
+```
+
+#### Modules Cleanup
+
+At the conclusion of a test file, Terraform will attempt to destroy every resource created during the execution of that test file. When alternate modules are loaded, the order in which objects are destroyed is important. For example, in our first [Modules](#modules) example earlier we cannot destroy the resources created in the "setup" `run` block before the objects created in the "execute" `run` block, as S3 bucket created in the "setup" step cannot be destroyed while it contains objects.
+
+Terraform will destroy resources in the following order, and this order is important as it may affect the structure of your testing files:
+
+1. Destroy the resources held in the main state file first, so you should not create resources in alternate modules that depend on resources from your main configuration.
+    - Note that data sources can refer to objects in your main configuration, as Terraform doesn't have to destroy data sources.
+2. Destroy the resources created by alternate modules in `run` block reverse order.
+    - From our example, any resources created in the "verify" `run` block would be destroyed before resources created in the "setup" `run` block. Note, that in our example this doesn't particularly matter as our "verify" `run` block only loads a data source and creates no resources.
```

Comment:
Can we move this to be a paragraph below? I think the thought stands well enough below the list, and it will leave the list looking smaller and more scannable.

Reply:
Also can you link back up to the example?